draft recommendation page

salehudeen opened the pull request:
- Add slight dynamic styling to all pages
- Add recommendation page
- Add scroll
- Style scroll bar

salehudeen committed:
draft recommendation page

David06A committed:
Fix: Overflow and text size/border

David06A committed:
Fix: Decrease card size

David06A committed:
Fix: Scroll overflow

David06A committed:
Fix: Question page height

David06A committed:
Fix: set height

David06A committed:
Fix: Improve scrollbar and add padding

SwyftPain commented on Front-End/src/pages/recommendation/RecommendationPage.tsx:
```diff
@@ -0,0 +1,68 @@
+    const [, recommended] = regex.exec(text);
+    const links = text.replace(recommended, "");
+    recommendedText = recommended;
+    linkText = links;
```
`linkText = links;`
I would still make these clickable, either via html formatting or the Link component, but just personal opinion.

David06A replied:
Agree

SwyftPain commented on Front-End/src/pages/recommendation/RecommendationPage.tsx:
```diff
@@ -0,0 +1,68 @@
+// handle data from backend
+
+const RecommendationPage = () => {
+    const [screenHeight, setScreenHeight] = useState<number>(
```
Not too big of a fan of setting the screen height directly like this as on different resolution monitors, may not scale too well. Personal opinion.

David06A replied:
Yeah its not good, but atm it doesn't scale at all without. Would need to work on a more dynamic solution in future

SwyftPain replied:
consider minHeight, maxHeight and fit-content maybe?

SwyftPain commented on Front-End/src/pages/recommendation/Recommendation.css:
```diff
@@ -0,0 +1,51 @@
+    margin: 20px;
+}
+.links {
+    /* display: flex;
```
Just in general, no need to leave in commented code with github due to the ability to see older versions of code and just adding on extra lines of code that may not be useful (due to the fact they are not being used, it is not JS-DOC, its just commented out CSS styles).

David06A replied:
Agree

SwyftPain replied:
Fixed.

SwyftPain commented on Front-End/src/pages/questions-page/QuestionPage.tsx:
```diff
@@ -23,6 +23,9 @@ const QuestionPage = ({ nextPage }: any) => {
         ProjectCreationQuestions
     );
+    const [screenHeight, setScreenHeight] = useState<number>(
+        window.innerHeight
```
I started commenting from the bottom, there is a comment below about this. About screen height.

SwyftPain commented on Front-End/src/pages/questions-page/QuestionPage.css:
```diff
@@ -1,57 +1,69 @@
-
```
This may just be my own ignorance, but why is the first line an empty line? That may just be how the github shows it when its  changed and in that case, it is my ignorance. Just wondering. This is not a critique.

SwyftPain replied:
It was my own ignorance.

SwyftPain commented on Front-End/src/pages/home-page/HomePage.tsx:
```diff
@@ -8,14 +8,20 @@ type HomePageProps = {
 export const HomePage = ({ nextPage }: HomePageProps) => {
     const [stacks, setStacks] = useState<Stack[]>(PopularStacks);
+    const [screenHeight, setScreenHeight] = useState<number>(
+        window.innerHeight
```
Comment below about screen height

SwyftPain commented on Front-End/src/pages/home-page/HomePage.css:
```diff
@@ -1,167 +1,179 @@
+    box-shadow: var(--bubbles) 2px 2px 10px;
 }
 
 /* CSS */
```
Isn't everything above also CSS? Also make indentation consistent or use CSS formatter.

SwyftPain replied:
Fixed.

SwyftPain commented on Front-End/src/App.tsx:
```diff
@@ -4,7 +4,7 @@ import "./App.css";
 import { HomePage } from "./pages/home-page/HomePage";
 import QuestionPage from "./pages/questions-page/QuestionPage";
-
+import RecommendationPage from "./pages/recommendation/RecommendationPage";
```
i would put a new line between import and the function just for the readability sake of it.
Separate by functionality.

SwyftPain replied:
Fixed.

SwyftPain reviewed (dismissed):
Check the comments. There may be some extra stuff I added or something I missed, but mainly readability related stuff. Functionality seems fine. But its easier to review if code is properly organized. This may not be fully your work that I commented on, but would change before deployment to production.

swyftpain96 commented on Front-End/src/pages/recommendation/RecommendationPage.tsx:
```diff
@@ -0,0 +1,68 @@
+// this regex function will split the text when it sees References:
+function separateText(text: any) {
+    const regex = /^(.*?)(?=\bReferences:)/is;
+    const [, recommended] = regex.exec(text);
```
Type 'RegExpExecArray | null' must have a '[Symbol.iterator]()' method that returns an iterator.
Just noticed this in my IDE.

SwyftPain btw, different acc in IDE.

swyftpain96 commented on Front-End/src/App.css:
```diff
@@ -1,296 +1,1291 @@
+    color: var(--text-secondary);
 }
 
 #stars {
```
This is a huge mess.

swyftpain96 commented on Front-End/src/index.css:
```diff
@@ -1,41 +1,39 @@
+    color-scheme: light dark;
+    color: rgba(255, 255, 255, 0.87);
+    background-color: #242424;
 
```
No need for empty line here

SwyftPain replied:
Fixed.

SwyftPain committed:
Fixed formatting/readability issues

- Formatted with prettier every file
- Removed empty lines where they don't belong
- Added line splits where they do belong
- Assigned missing type declarations that were defined but not assigned
- Made indentation consistent with prettier
- Left comments about portions i am unable to do/unsure how

SwyftPain commented:
I added a commit, majority of it should be improved, ill let you guys check it out and approve it, so I am not biased. Removed my own review.

David06A committed:
Fix: Removed spaces from formatting

David06A reviewed (approved):
Overall code quality, could be improved. But, for the sake of the demo being tonight. We will introduce some tech debt.

Next project should be cleaner as we will have more contributors, so there will be less of a rush. Before the demo / I tweet I will tidy up the front end to make it look slightly more appealing UI.
